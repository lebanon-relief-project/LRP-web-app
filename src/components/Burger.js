import React from "react";
import styled from "styled-components";
import { bool, func } from "prop-types";

const Burger = ({ open, setOpen, ...props }) => {
  const isExpanded = open ? true : false;

  return (
    <StyledBurger
      aria-label="Toggle menu"
      aria-expanded={isExpanded}
      open={open}
      onClick={() => setOpen(!open)}
      {...props}
    >
      <span />
      <span />
      <span />
    </StyledBurger>
  );
};

Burger.propTypes = {
  open: bool.isRequired,
  setOpen: func.isRequired,
};

const StyledBurger = styled.button`
  @media (min-width: 576px) {
    display: none;
  }

  position: absolute;
  top: 5%;
  right: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
  span {
    width: 2rem;
    height: 0.25rem;
    background: ${({ open }) => (open ? "black" : "black")};
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;
    :first-child {
      transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
    }
    :nth-child(2) {
      opacity: ${({ open }) => (open ? "0" : "1")};
      transform: ${({ open }) => (open ? "translateX(20px)" : "translateX(0)")};
    }
    :nth-child(3) {
      transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }
`;

export default Burger;
