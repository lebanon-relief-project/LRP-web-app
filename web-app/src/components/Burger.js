import React from "react";
import styled from "styled-components";

const Burger = ({ open, setOpen, ...props }) => {
  const isExpanded = open ? true : false;

  return (
    <StyledBurger
      aria-label="Toggle menu"
      aria-expanded={isExpanded}
      open={open}
      data-testid="burgerButton"
      onClick={() => setOpen(!open)}
      {...props}
    >
      <span />
      <span />
      <span />
    </StyledBurger>
  );
};

const StyledBurger = styled.button`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  position: fixed;

  right: 2rem;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;

  span {
    width: 2rem;
    height: 0.25rem;
    background: black;
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;

    transform-origin: 1px;
    :first-child {
      transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
    }
    :nth-child(2) {
      opacity: ${({ open }) => (open ? "0" : "1")};
      transform: ${({ open }) => (open ? "translateX(20px)" : "translateX(0)")};
    }
    :nth-child(3) {
      transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }
`;

export default Burger;
