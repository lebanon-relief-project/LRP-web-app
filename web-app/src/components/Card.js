import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import devices from "../styles/Devices";
import colours from "../styles/Colours";

const Card = (props) => {
  const { children, title, path, buttonText, ...rest } = props;
  return (
    <StyledCard {...rest}>
      <CardTitle {...rest}>{title}</CardTitle>
      {children}

      <br />
      {props.externalPath ? (
        <StyledAnchor
          href={props.externalPath}
          target="_blank"
          rel="noopener noreferrer"
        >
          {buttonText}
        </StyledAnchor>
      ) : (
        <StyledLink {...rest} to={path}>
          {buttonText}
        </StyledLink>
      )}
    </StyledCard>
  );
};

const StyledAnchor = styled.a`
  position: ${(props) => (props.secondary ? "static" : "absolute")};
  bottom: 0;
  left: 50%;
  -webkit-transform: translateX(${(props) => (props.secondary ? "0" : "-50%")});
  background-color: ${(props) =>
    props.disabled ? `${colours.grey}` : `${colours.yellow}`};
  color: black;
  font-style: normal;
  font-weight: bold;
  padding: 0.5rem;
  justify-content: center;
  display: flex;
  width: ${(props) => (props.secondary ? "30%" : "90%")};
  @media (max-width: ${devices.mobile}) {
    width: ${(props) => (props.secondary ? "50%" : "90%")};
  }
  margin-bottom: 5%;
`;

const StyledLink = styled(Link)`
  position: ${(props) => (props.secondary ? "static" : "absolute")};
  bottom: 0;
  left: 50%;
  -webkit-transform: translateX(${(props) => (props.secondary ? "0" : "-50%")});
  background-color: ${(props) =>
    props.disabled ? `${colours.grey}` : `${colours.yellow}`};
  color: black;
  font-style: normal;
  font-weight: bold;
  justify-content: center;
  display: flex;
  text-align: center;
  min-width: ${(props) => (props.secondary ? "25%" : "90%")};
  width: 50%;
  @media (max-width: ${devices.mobile}) {
    width: ${(props) => (props.secondary ? "50%" : "90%")};
    max-width: ${(props) => (props.secondary ? "100%" : "90%")};
  }
  margin-bottom: 5%;
  padding: 0.5rem;
`;

const CardTitle = styled.legend`
  font-family: Playfair Display;
  font-weight: ${(props) => (props.secondary ? 900 : "bold")};
  font-size: ${(props) => (props.secondary ? "38px" : "24px")};
  min-width: auto;
  line-height: 100%;
  padding: ${(props) => (props.secondary ? "0" : "0 0.5rem")};
  text-align: ${(props) => (props.secondary ? "left" : "center")};
  margin-bottom: 0;
  @media (max-width: ${devices.mobile}) {
    font-size: 30px;
    line-height: 38px;
  }
`;

const StyledCard = styled.fieldset`
  position: relative;
  border: ${(props) =>
    props.secondary ? "none" : `1px solid ${colours.blue}`};
  width: ${(props) => (props.secondary ? "100%" : "25%")};
  padding: ${(props) => (props.secondary ? "1rem 0" : "1rem 1.2rem")};
  margin: 3rem 1rem;
  height: 20%;
  font-family: Raleway;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0em;
  text-align: left;
  min-width: 360px;

  @media (max-width: ${devices.ipadpro}) {
    width: ${(props) => (props.secondary ? "100%" : "40%")};
    min-width: 0;
  }
  @media (max-width: ${devices.ipad}) {
    margin: ${(props) => (props.secondary ? "3rem 0" : "3rem 1rem")};
  }
  @media (max-width: ${devices.mobile}) {
    width: 90%;
    margin: 1rem auto;
  }
`;

export default Card;
