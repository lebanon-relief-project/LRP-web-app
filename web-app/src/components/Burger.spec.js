import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Card from "./Card";
import { MemoryRouter } from "react-router-dom";
import "jest-styled-components";
import Burger from "./Burger";
import { act } from "react-dom/test-utils";

const setOpenMock = jest.fn();

afterEach(() => {
  jest.clearAllMocks();
});

describe("the Burger component", () => {
  let container;
  it("should match the snapshot when open is set to false", () => {
    ({ container } = renderBurger(false, setOpenMock));

    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when open is set to true", () => {
    ({ container } = renderBurger(true, setOpenMock));

    expect(container).toMatchSnapshot();
  });

  it("should execute setOpenMock only when burger button is clicked and open is set to false", () => {
    let getByTestId;
    ({ container, getByTestId } = renderBurger(false, setOpenMock));

    expect(setOpenMock).toHaveBeenCalledTimes(0);

    act(() => {
      fireEvent.click(getByTestId("burgerButton"));
    });

    expect(setOpenMock).toHaveBeenCalledTimes(1);
    expect(setOpenMock).toHaveBeenCalledWith(true);
  });

  it("should execute setOpenMock only when burger button is clicked and open is set to true", () => {
    let getByTestId;
    ({ container, getByTestId } = renderBurger(true, setOpenMock));

    expect(setOpenMock).toHaveBeenCalledTimes(0);

    act(() => {
      fireEvent.click(getByTestId("burgerButton"));
    });

    expect(setOpenMock).toHaveBeenCalledTimes(1);
    expect(setOpenMock).toHaveBeenCalledWith(false);
  });
});

const renderBurger = (open, setOpenMock) => {
  return render(<Burger open={open} setOpen={setOpenMock} />);
};
