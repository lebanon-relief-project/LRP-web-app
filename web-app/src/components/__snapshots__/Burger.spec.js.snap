// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Burger component should match the snapshot when open is set to false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  position: fixed;
  right: 2rem;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  outline: none;
}

.c0 span {
  width: 2rem;
  height: 0.25rem;
  background: #002766;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 span:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 span:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c0 span:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<div>
  <button
    aria-expanded="false"
    aria-label="Toggle menu"
    class="c0"
    data-testid="burgerButton"
  >
    <span />
    <span />
    <span />
  </button>
</div>
`;

exports[`the Burger component should match the snapshot when open is set to true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  position: fixed;
  right: 2rem;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  outline: none;
}

.c0 span {
  width: 2rem;
  height: 0.25rem;
  background: #002766;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 span:first-child {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 span:nth-child(2) {
  opacity: 0;
  -webkit-transform: translateX(20px);
  -ms-transform: translateX(20px);
  transform: translateX(20px);
}

.c0 span:nth-child(3) {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div>
  <button
    aria-expanded="true"
    aria-label="Toggle menu"
    class="c0"
    data-testid="burgerButton"
    open=""
  >
    <span />
    <span />
    <span />
  </button>
</div>
`;
